libs/platform/ebpf_epoch.c

```diff
@@ -122,9 +121,48 @@ _ebpf_epoch_get_release_epoch(_Out_ int64_t* released_epoch);
 static void
 _ebpf_flush_worker(_In_ void* context);
 
+static void
+_ebpf_epoch_stale_worker(_In_ void* work_item_context, _In_ void* parameter_1)
+{
+    UNREFERENCED_PARAMETER(work_item_context);
+    UNREFERENCED_PARAMETER(parameter_1);
+    ebpf_epoch_enter();
+    ebpf_epoch_exit();
+}
+
 ebpf_result_t
 _ebpf_epoch_update_thread_state(uint32_t cpu_id, uintptr_t thread_id, int64_t current_epoch, bool enter);
 
+typedef enum _ebpf_epoch_per_cpu_flags
+{
+    EBPF_EPOCH_PER_CPU_TIMER_ARMED = (1 << 0),
+    EBPF_EPOCH_PER_CPU_STALE = (1 << 1),
+} ebpf_epoch_per_cpu_flags_t;
+
+static bool
+_ebpf_get_per_cpu_flag(_In_ const ebpf_epoch_cpu_entry_t* cpu_entry, ebpf_epoch_per_cpu_flags_t flag)
+{
+    return (cpu_entry->flags & flag) != 0;
+}
+
+static void
+_ebpf_set_per_cpu_flag(_In_ ebpf_epoch_cpu_entry_t* cpu_entry, ebpf_epoch_per_cpu_flags_t flag, bool state)
+{
```

Comment:
consider using https://docs.microsoft.com/en-us/windows-hardware/drivers/ddi/wdm/nf-wdm-interlockedor

Reply:
Will fix this in another PR.